### PR TITLE
feat: add lucia auth option

### LINKED ISSUE #1606: feat: adding Lucia for Authentication
### Is your feature request related to a problem? Please describe.

not a problem, but it would be great to have another selection for [Lucia](https://lucia-auth.com/)
users would be able to choose between Auth.js/Lucia for handling their Auth

### Describe the solution you'd like to see

a option to select from either Lucia or Auth.js (NextAuth.js)

### Describe alternate solutions

N/A

### Additional information

I would be happy to do it 😃

### PR DESCRIPTION
**Pull request by ronanru**
Closes #1606
Related to #1593

Lucia Auth is a popular alternative to NextAuth, and I think it will be a great addition to ct3a

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

- Added an option for Lucia Auth

---

## Screenshots

image

💯

### CODE REVIEW

**Comment by ronanru** [comment]
I manually tested all possible combinations and didn't find any bugs.

File: cli/template/extras/src/app/_components/logout-button-trpc.tsx
Comment (juliusmarminge) [inline cli/template/extras/src/app/_components/logout-button-trpc.tsx:1]: trpc seems redundant here? Simpler just using a regular form?

```tsx
<form action="/api/auth/logout" method="post">
  <button type="submit">Log out</button>
</form>
```

File: cli/template/extras/src/app/_components/logout-button.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+"use client";
+
+import { useRouter } from "next/navigation";
+
+export function LogOutButton({ className }: { className?: string }) {
```
Comment (juliusmarminge) [inline cli/template/extras/src/app/_components/logout-button.tsx:5]: Same as above

File: cli/template/extras/src/app/api/auth/discord/signin/route.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  }
+  const [url, state] = await discordAuth.getAuthorizationUrl();
+  const cookieStore = context.cookies();
+  cookieStore.set("discord_oauth_state", state, {
+    httpOnly: true,
+    secure: env.NODE_ENV === "production",
+    path: "/",
+    maxAge: 60 * 60,
+  });
+  return new Response(null, {
```
Comment (juliusmarminge) [inline cli/template/extras/src/app/api/auth/discord/signin/route.ts:28]: `NextResponse.redirect()` ?

File: cli/template/extras/src/app/page/with-lucia-trpc-tw.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import Link from "next/link";
+
+import { CreatePost } from "~/app/_components/create-post";
+import { LogOutButton } from "~/app/_components/logout-button";
+import { getPageSession } from "~/server/auth";
```
Comment (juliusmarminge) [inline cli/template/extras/src/app/page/with-lucia-trpc-tw.tsx:5]: Maybe use a same-named export for both NextAuth and lucia so we can use the same frontend stuff for both?

**Comment by juliusmarminge** [comment]
Also enable CI for lucia - since you have PR'd from next (default branch) I can't push to your branch

General diff:

CleanShot 2023-10-21 at 18 11 14@2x

**Comment by dominik-sfl** [comment]
is this happening? would love to see this integrated.

**Comment by juliusmarminge** [comment]
Hey @ronanru, thank you for putting in the effort for this and the db option PR.

However, we have decided not to add Lucia as an auth option as we don't see the value it adds over next-auth, especially with it's auth.js port around the corner with v5.

Me and @c-ehrlich will try and get the db option PR merged soon though!

**Pull request closed without merging**